### PR TITLE
Add CheckIpAddressAvailability API

### PR DESCRIPTION
**Pull request by AhmedSalama-MS**
Add CheckIpAddressAvailability API on Virtual Networks

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@AhmedSalama-MS__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ahsalam). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: arm-network/2016-06-01/swagger/network.json [hunk 1]
```diff
@@ -4251,6 +4251,49 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{virtualNetworkName}/CheckIpAddressAvailability": {
+      "get": {
+        "operationId": "CheckIpAddressAvailability",
```
Comment (amarzavery) [inline arm-network/2016-06-01/swagger/network.json]: This should be "VirtualNetworks_CheckIpAddressAvailability" as the operation is happening on a virtualNetwork.
CheckDnsNameAvailability is at a higher level if you compare the path for both the operations:
"/subscriptions/{subscriptionId}/providers/Microsoft.Network/locations/{location}/CheckDnsNameAvailability"

File: arm-network/2016-06-01/swagger/network.json [hunk 2]
```diff
@@ -4251,6 +4251,49 @@
...
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
```
Comment (amarzavery) [inline arm-network/2016-06-01/swagger/network.json]: Please make sure the description is not empty anywhere in the spec.

Comment (amarzavery) [inline arm-network/2016-06-01/swagger/network.json]: The description becomes the clientside doc and the msdn rest api doc.

File: arm-network/2016-06-01/swagger/network.json [hunk 3]
```diff
@@ -4251,6 +4251,49 @@
...
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{virtualNetworkName}/CheckIpAddressAvailability": {
+      "get": {
+        "operationId": "CheckIpAddressAvailability",
+        "description": "Checks whether a private Ip address is available for use.",
+        "parameters": [
+          {
+            "name": "IpAddress",
```
Comment (DeepakRajendranMsft) [inline arm-network/2016-06-01/swagger/network.json]: ipAddress

File: arm-network/2016-06-01/swagger/network.json [hunk 4]
```diff
@@ -4251,6 +4251,49 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworks/{virtualNetworkName}/CheckIpAddressAvailability": {
+      "get": {
+        "operationId": "CheckIpAddressAvailability",
+        "description": "Checks whether a private Ip address is available for use.",
```
Comment (DeepakRajendranMsft) [inline arm-network/2016-06-01/swagger/network.json]: private IP address

File: arm-network/2016-06-01/swagger/network.json [hunk 5]
```diff
@@ -4251,6 +4251,49 @@
...
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/IPAddressAvailabilityResult"
```
Comment (DeepakRajendranMsft) [inline arm-network/2016-06-01/swagger/network.json]: PrivateIPAddressAvailabilityResult

File: arm-network/2016-06-01/swagger/network.json [hunk 6]
```diff
@@ -7965,6 +8008,22 @@
       },
       "description": "Response for CheckDnsNameAvailability Api servive call"
     },
+    "IPAddressAvailabilityResult": {
+      "properties": {
+        "available": {
+          "type": "boolean",
+          "description": "Private Ip address availability (True/False)"
```
Comment (DeepakRajendranMsft) [inline arm-network/2016-06-01/swagger/network.json]: True/False is not required

File: arm-network/2016-06-01/swagger/network.json [hunk 7]
```diff
@@ -7965,6 +8008,22 @@
...
+        },
+        "availableIpAddresses": {
```
Comment (DeepakRajendranMsft) [inline arm-network/2016-06-01/swagger/network.json]: availableIPAddresses

File: arm-network/2016-06-01/swagger/network.json [hunk 8]
```diff
@@ -7965,6 +8008,22 @@
...
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Other available private Ip addresses if the asked for address is taken"
```
Comment (DeepakRajendranMsft) [inline arm-network/2016-06-01/swagger/network.json]: Contains other available private IP addresses

File: arm-network/2016-06-01/swagger/network.json [hunk 9]
```diff
@@ -7965,6 +8008,22 @@
...
+        }
+      },
+      "description": "Response for CheckIpAddressAvailability Api servive call"
```
Comment (DeepakRajendranMsft) [inline arm-network/2016-06-01/swagger/network.json]: CheckIPAddressAvailability

File: arm-network/2016-06-01/swagger/network.json (hunk 5 above)
Comment (AhmedSalama-MS) [inline arm-network/2016-06-01/swagger/network.json]: Will leave this as is to match NRP's model naming.

**Comment by DeepakRajendranMsft** [comment]
LGTM

**Pull request merged**